### DISCUSSION TITLE
Having problems trying to use duckdb-async

### DISCUSSION THREAD

**Discussion by Kennyac** [vercel/next.js#49709] [Help]
### Summary

I have just started to use next.js

I am running mainly on windows but I have tried duckdb-async with next.js on both ubuntu and windows

I am using node version 18.16.0

I have tried duckdb-async in fastify and express and it worked

I have added duckdb-async and webpack to my project next.js project 

From the web pack documentation (https://nextjs.org/docs/app/api-reference/next-config-js/webpack) its not clear what I need to configure to make it work

I recieve the following  error

- error ./node_modules/@mapbox/node-pre-gyp/lib/util/nw-pre-gyp/index.html
Module parse failed: Unexpected token (1:0)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
> <!doctype html>
| <html>
| <head>

Import trace for requested module:
./node_modules/@mapbox/node-pre-gyp/lib/util/nw-pre-gyp/index.html
./node_modules/@mapbox/node-pre-gyp/lib/ sync ^\.\/.*$
./node_modules/@mapbox/node-pre-gyp/lib/node-pre-gyp.js
./node_modules/duckdb/lib/duckdb-binding.js
./node_modules/duckdb/lib/duckdb.js
./node_modules/duckdb-async/dist/duckdb-async.js
./src/app/api/duckdb/route.ts

### Additional information

```js
to reproduce 
yarn install
yarn dev
browse to http://localhost:3001/api/duckdb
```

### Example

https://drive.google.com/file/d/1dceXEXiaH01-SJKxu2ckgoWCTf8f79Rm/view?usp=sharing

**Comment by pgzmnk** [comment]
Not sure about your specific issue. 

Here's a base template for DuckDB-Wasm shell with Next.js. Note the webpack config in `next.config.js`.

https://github.com/pgzmnk/nextjs-typescript/blob/main/next.config.js

**Reply by Kennyac** [reply]
Thanks 
i give that a try but i presume that would only work  client end code

I was wanting to use duckdb on the server; would that still work is there likely to be a cost for running on server

I am new to this is there any documentation that would detail what this code is doing

**Comment by geoffsee** [comment]
nextjs config now has a transpileModules prop which accepts string[]. Simply add the duckdb packages' names in there. 

Hope this helps!

**Comment by Rosslington** [comment]
Did you ever get around this?

**Reply by Kennyac** [reply]
Gave up with next.js and created a fastify service to wrap what I wanted form duck db ; 

I want what Iam doing to be run on webserver never on the client

Think keeping it separate is better anyway

**Comment by srossross** [comment]
Not very helpful, but this is almost solved in the new the unreleased nextjs@15 

In `next.config.mjs` you can now do 

```
/** @type {import('next').NextConfig} */

const nextConfig = {
  serverExternalPackages: ["duckdb"],
};

export default nextConfig;
``` 

However, this does not work with `duckdb-async`

**Reply by pencilcheck** [reply]
Or if you are on nextjs 14, try this

```
  experimental: {
    serverComponentsExternalPackages: ["duckdb", "duckdb-async"],
  },
```

And btw, duckdb-async works with this configuration above on nextjs 14, it is not perfect as some parsers aren't perfect but it executes

**Comment by NateV** [comment]
I'd love to know _why_ this problem happens. Why are nextjs and duckdb incompatible in this way? Knowing more might help figure out workarounds.

**Reply by pencilcheck** [reply]
@NateV read my response above to @srossross it worked on nextjs 14

**Comment by Sylvestre67** [comment]
If you are on nextjs 15, this is the way to go: 
```typescript
const nextConfig = {
   ...
    serverExternalPackages: ["duckdb", "duckdb-async"],
};
```
